apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletBalancesAlph.ts

```diff
@@ -16,30 +16,60 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { useQueries } from '@tanstack/react-query'
+import { AddressHash } from '@alephium/shared'
+import { useQueries, UseQueryResult } from '@tanstack/react-query'
 
-import { DataHook, SkipProp } from '@/api/apiDataHooks/apiDataHooksTypes'
+import { SkipProp } from '@/api/apiDataHooks/apiDataHooksTypes'
+import { combineIsLoading } from '@/api/apiDataHooks/apiDataHooksUtils'
 import { combineBalances } from '@/api/apiDataHooks/wallet/combineBalances'
-import { useFetchWalletLastTransactionHashes } from '@/api/apiDataHooks/wallet/useFetchWalletLastTransactions'
-import { addressAlphBalancesQuery } from '@/api/queries/addressQueries'
+import { useFetchWalletUpdatesSignals } from '@/api/apiDataHooks/wallet/useFetchWalletLastTransactions'
+import { addressAlphBalancesQuery, AddressAlphBalancesQueryFnData } from '@/api/queries/addressQueries'
 import { useAppSelector } from '@/hooks/redux'
 import { DisplayBalances } from '@/types/tokens'
 
-const useFetchWalletBalancesAlph = (props?: SkipProp): DataHook<DisplayBalances | undefined> => {
-  const { data: latestTxHashes, isLoading: isLoadingLatestTxHashes } = useFetchWalletLastTransactionHashes(props)
+// Using undefined to avoid adding noUncheckedIndexedAccess in tsconfig while maintaining strong typing when accessing
+// values through indexes, ie: alphBalances[addressHash]
+export interface AddressesAlphBalances {
+  data: Record<AddressHash, DisplayBalances | undefined>
+  isLoading: boolean
+}
+
+export const useFetchWalletBalancesAlph = (props?: SkipProp) =>
+  useFetchWalletBalancesAlphBy({ combine: combineBalances, skip: props?.skip })
+
+export const useFetchWalletBalancesAlphByAddress = (props?: SkipProp) =>
+  useFetchWalletBalancesAlphBy({ combine: combineBalancesByAddress, skip: props?.skip })
+
+interface UseFetchWalletBalancesAlphProps<T> extends SkipProp {
+  combine: (results: UseQueryResult<AddressAlphBalancesQueryFnData>[]) => { data: T; isLoading: boolean }
+}
+
+const useFetchWalletBalancesAlphBy = <T>({ combine, skip }: UseFetchWalletBalancesAlphProps<T>) => {
```

Comment:
Why keeping the "By" at the end?

Reply:
Because there is `useFetchWalletBalancesAlph` already

Reply:
`useFetchWalletBalancesAlphBase`, `useFetchWalletBalancesAlphRaw`?

Reply:
How about we do the same we did for tokens where we have:
```ts
// Base hook, not exported
const useFetchWalletBalancesTokens

// exported hooks
export const useFetchWalletBalancesTokensArray
export const useFetchWalletBalancesTokensByToken
export const useFetchWalletBalancesTokensByAddress
```

so that'd mean

```ts
// Base hook, not exported
const useFetchWalletBalancesAlph

// exported hooks
export const useFetchWalletBalancesAlphArray
export const useFetchWalletBalancesAlphByAddress
```

Reply:
Implemented in https://github.com/alephium/alephium-frontend/pull/869/commits/bd35c6ef3e8813f9d8df6e7695abc8e68dbe4883